hail_search/queries/base.py

```diff
@@ -354,67 +363,64 @@ def _filter_entries_table(self, ht, sample_data, inheritance_mode=None, inherita
-        sample_index_id_map = hl.dict(sample_index_id_map)
-        sample_individual_map = {s['sample_id']: s['individual_guid'] for s in sample_data}
-        missing_samples = set(sample_individual_map.keys()) - set(sample_id_index_map.keys())
+
+        missing_samples = set()
+        family_sample_index_data = []
+        family_guids = sorted(sample_data.keys())
+        for family_guid in family_guids:
+            samples = sample_data[family_guid]
+            ht_family_samples = ht_globals.family_samples[family_guid]
+            missing_family_samples = [s['sample_id'] for s in samples if s['sample_id'] not in ht_family_samples]
+            if missing_family_samples:
+                missing_samples.update(missing_family_samples)
+            else:
+                family_sample_index_data.append(
+                    (ht_globals.family_guids.index(family_guid), [
+                        (ht_family_samples.index(s['sample_id']), cls._sample_entry_data(s, family_guid, ht_globals))
+                        for s in samples
+                    ])
+                )
+
         if missing_samples:
             raise HTTPBadRequest(
                 reason=f'The following samples are available in seqr but missing the loaded data: {", ".join(sorted(missing_samples))}'
             )
 
-        affected_id_map = {AFFECTED: AFFECTED_ID, UNAFFECTED: UNAFFECTED_ID, UNKNOWN_AFFECTED: UNKNOWN_AFFECTED_ID}
-        sample_index_affected_status = hl.dict({
-            sample_id_index_map[s['sample_id']]: affected_id_map.get(s['affected']) for s in sample_data
-        })
-        sample_index_individual_map = hl.dict({
-            sample_id_index_map[sample_id]: i_guid for sample_id, i_guid in sample_individual_map.items()
-        })
-        sample_id_family_map = {s['sample_id']: s['family_guid'] for s in sample_data}
-        sample_index_family_map = hl.dict({sample_id_index_map[k]: v for k, v in sample_id_family_map.items()})
-        family_guids = sorted(set(sample_id_family_map.values()))
-        family_index_map = {f: i for i, f in enumerate(family_guids)}
-        num_families = len(family_index_map)
-        family_sample_indices = [None] * num_families
-        sample_id_family_index_map = {}
-        for sample_id, family_guid in sorted(sample_id_family_map.items()):
-            sample_index = sample_id_index_map[sample_id]
-            family_index = family_index_map[family_guid]
-            if not family_sample_indices[family_index]:
-                family_sample_indices[family_index] = []
-            sample_id_family_index_map[sample_id] = (family_index, len(family_sample_indices[family_index]))
-            family_sample_indices[family_index].append(sample_index)
-        family_sample_indices = hl.array(family_sample_indices)
+        sorted_family_sample_data = [[hl.eval(sample) for _, sample in samples] for _, samples in family_sample_index_data]
```

Comment:
This `hl.eval` in a loop looks like it's gonna be slow.  iiuc, `family_sample_index_data` is a mapping of the index of the family_guid to a list of the indexes of the samples?  is it not already `eval`ed?

Reply:
so they are a series of structs and somehow they get like un-evaled and you can't use them directly later on without evaling them. they are just a flat array of structs they aren't like a table on disk so the eval is pretty fast, but I can probably get it to work without it